submissions/servers/servers.md

```diff
@@ -0,0 +1,47 @@
+
+{% include image.html src='https://spectacularagency.files.wordpress.com/2010/09/lockstep.jpg' %}
+
+A [Fully Deterministic algorithm](https://en.wikipedia.org/wiki/Deterministic_algorithm) is also used in Factorio. A such algorithm will produce the same output when given the same input. This means that there is no randomness in the results, which is very useful in some cases such as Factorio. A Fully Deterministic algorithm is defined by the following:
+- It must not use any other data other than the input to the algorithm. Disallowed is: random numbers, stored disk data, global variables, timers (i.e. from when the program is running)
+- The algorithm must operate in a way that is not time-sensitive. An example of the opposite of this would be if multiple instances of a program were writing into an Excel spreadsheet and another program would read the last line of the sheet. This would make the program time-sensitive as if one instance of the writing programs is delayed by a few seconds, it can produce a completely different order of the Excel rows, leaving the program that reads the last line with a completely differen input
+As explained in [Therenas's cover on 1.1 optimizations](https://alt-f4.blog/ALTF4-15/#how-does-this-optimization-work-on-a-technical-level), if things were all multithreaded, the order of things updating wouldn't work exactly as it should, which makes the game updated partially Fully Deterministic algorithms from the second definition! The main part of that article however stops the *whole update cycle* from being a Fully Deterministic algorithm as belts are now multithreaded - the order in which they update is not guaranteed.
```

Comment:
You need to add, that they figured out how to make the belts deterministic again - after multithreading - by putting it together in a smart way. You might want to reference the newest FFF too, where boskid talks about multi-threading again.